spring/boot3-autoconfigure/src/main/java/com/linecorp/armeria/internal/spring/ArmeriaConfigurationUtil.java

```diff
@@ -214,6 +208,48 @@ public static void configureServerWithArmeriaSettings(
         armeriaServerBuilderConsumers.forEach(consumer -> consumer.accept(server));
     }
 
+    private static void configureDependencyInjector(
+            ServerBuilder server, ArmeriaSettings settings,
+            List<DependencyInjector> dependencyInjectors, BeanFactory beanFactory) {
+
+        DependencyInjector dependencyInjector = null;
+        if (!dependencyInjectors.isEmpty()) {
+            // dependencyInjectors from beans are added first, which will be used first thing,
+            // then from armeriaServerConfigurators and armeriaServerBuilderConsumers.
+            for (DependencyInjector injector : dependencyInjectors) {
+                if (dependencyInjector == null) {
+                    dependencyInjector = injector;
+                } else {
+                    dependencyInjector.orElse(injector);
+                }
+            }
+        }
+
+        if (settings.isEnableAutoInjection()) {
+            if (dependencyInjector == null) {
+                dependencyInjector = SpringDependencyInjector.of(beanFactory);
+            } else {
+                dependencyInjector = dependencyInjector.orElse(SpringDependencyInjector.of(beanFactory));
+            }
+        }
+
+        if (settings.getAthenz() != null) {
+            final String athenzServiceClass = "com.linecorp.armeria.server.athenz.AthenzService";
+            if (hasClass(athenzServiceClass)) {
+                dependencyInjector = AthenzSupport.injectAthenzDecorator(settings.getAthenz(),
```

Comment:
Note) Understood that the `AthenzDecorator` is injected as the first `dependencyInjector` (e.g. has higher priority over user-registered injectors)